store/sqlstore/thread_store.go

```diff
@@ -224,16 +226,14 @@ func (s *SqlThreadStore) GetTotalUnreadUrgentMentions(userId, teamId string, opt
 		Select("COALESCE(SUM(ThreadMemberships.UnreadMentions),0)").
 		From("ThreadMemberships").
 		Join("PostsPriority ON PostsPriority.PostId = ThreadMemberships.PostId").
+		Join("Threads ON Threads.PostId = ThreadMemberships.PostId").
+		Join("ChannelMembers ON ChannelMembers.UserId = ThreadMemberships.UserId AND ChannelMembers.ChannelId = Threads.ChannelId").
 		Where(sq.Eq{
 			"ThreadMemberships.UserId":    userId,
 			"ThreadMemberships.Following": true,
 			"PostsPriority.Priority":      model.PostPriorityUrgent,
 		})
 
-	if teamId != "" || !opts.Deleted {
-		query = query.Join("Threads ON Threads.PostId = ThreadMemberships.PostId")
-	}
-
 	if teamId != "" {
```

Comment:
The API doesn't allow an empty `teamId` when fetching user threads.
AFAICT we only pass an empty string as a `teamId` upon testing.